### ISSUE TITLE
linter: no-shadow: False positive for enums

### ISSUE THREAD

**Issue by keegan-lillo** [oxc-project/oxc#21604]
### What version of Oxlint are you using?

1.60.0

### What command did you run?

npx oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "categories": {
    "correctness": "error"
  },
  "env": {
    "builtin": true
  },
  "rules": {
    "no-shadow": "error"
  }
}

```

### What happened?

I get the following false positive when working with enums that have members with the same name as another variable in scope

```ts
export const Priority = 1234

export enum SomeEnum {
  Priority = 'PRIORITY',
}
```

```
  × eslint(no-shadow): 'Priority' is already declared in the upper scope.
   ╭─[file:///Users/keegan/repos/personal/oxlint-test/index.ts:1:14]
 1 │ export const Priority = 1234
   ·              ────┬───
   ·                  ╰── shadowed declaration is here
 2 │ 
 3 │ export enum SomeEnum {
 4 │   Priority = 'PRIORITY',
   ·   ──────────┬──────────
   ·             ╰── 'Priority' is declared here
 5 │ }
   ╰────
  help: Consider renaming 'Priority' to avoid shadowing the variable from the outer scope.
```

Repro: [no-shadow.zip](https://github.com/user-attachments/files/26920171/no-shadow.zip)

**Comment by camc314** [comment]
Hi, this is working as intended. please see https://typescript-eslint.io/rules/no-shadow/#why-does-the-rule-report-on-enum-members-that-share-the-same-name-as-a-variable-in-a-parent-scope

**Comment by keegan-lillo** [comment]
Wow. You learn something new every day! Thanks for super fast response.

**Comment by camc314** [comment]
It's not the most intuitive behaviour!

I've added an extra note to the diagnostic to help discoverability and comprehension of this behaviour:

https://github.com/oxc-project/oxc/pull/21607
